.github/workflows/build-docker-image.yaml

```diff
@@ -0,0 +1,155 @@
+name: Build Docker Image
+
+defaults:
+  run:
+    shell: bash
+
+on:
+  # Triggers when code is pushed to "main" branch
+  # This should only happen when PRs are merged
+  push:
+    branches:
+      - main
+
+  # Triggers the workflow when pull request created and updated
+  pull_request:
+    # Sequence of patterns matched against refs/heads
+    branches:
+      - main
+
+  # Triggers when a new release is created
+  release:
+    types:
+      - published
+
+  # Allows run of this workflow manually from the Actions tab
+  workflow_dispatch:
+
+
+env:
+  REGISTRY: ghcr.io
+  TEST_TAG: geoips:test
+
+jobs:
+  build_docker_image:
+    runs-on: ${{ vars.RUNNER }}
+    # Can't write to registry without this
+    permissions:
+      contents: read
+      packages: write
+    outputs:
+      IMAGE_NAME: ${{ steps.set_image_name.outputs.IMAGE_NAME }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup BuildX
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set the image name
+        id: set_image_name
+        run: |
+          image_name=${REGISTRY}/${GITHUB_REPOSITORY,,}
+          echo "IMAGE_NAME=${image_name}" >> ${GITHUB_ENV}
+          echo "IMAGE_NAME=${image_name}" >> ${GITHUB_OUTPUT}
+
+      - name: Login to the Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ secrets.GHCR_USER }}
+          password: ${{ secrets.GHCR_TOKEN }}
+
+      - name: Debug
+        run: |
+          echo $PWD
+          ls -la --color
+          git status
+
+      - name: Build and Push image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ env.IMAGE_NAME }}:${{ github.sha }}
+            ${{ env.IMAGE_NAME }}:test
+          file: "Dockerfile"
+          cache-from: type=registry,ref=${{ env.IMAGE_NAME }}:test
+          cache-to: type=inline
+
+  build_sphinx_html:
+    runs-on: ${{ vars.RUNNER }}
+    needs: build_docker_image
+    container:
+      image: ${{ needs.build_docker_image.outputs.IMAGE_NAME }}:${{ github.sha }}
+      credentials:
+        username: ${{ secrets.GHCR_USER }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Configure Git safe directory
+        run: git config --global --add safe.directory $PWD
+
+      - name: Install geoips documentation packages
+        run: pip install -e .[doc]
+
+      # - name: Download Docker image artifact
```

Comment:
Doesn't necessitate a review, but double check whether or not this commented out block needs to be included for a reason; otherwise remove before merge